### PR TITLE
Move functions in locallib.php to a class object groups

### PR DESCRIPTION
**Pull request by kei178**
## Description of the change

Moved the following functions in locallib.php to a new class object `groups.php`:

- `accredible_get_groups` -> `groups->get_groups`
- `accredible_get_templates` -> `groups->get_templates`
- `sync_course_with_accredible` -> `groups->sync_group_with`

## Coding style

The new class `groups.php` meets the Moodle coding style except for `Variable "apiRest" must be all lower-case` which will be addressed when refactoring `apiRest.php`.

## How to test

The new functions are covered by unit tests. You can also manually test the functions by the following steps.

### `groups->get_groups`

1. Go to the Accredible activity edit page.
2. Check if select options of "Cohort name (from dashboard)" are properly displayed.

### `groups->get_templates`

1. Temporarily update the `if` condition at Line116 in `mod_form.php` to return TRUE.
2. Go to the Accredible activity edit page.
3. Check if select options of "Group" are properly displayed.

#### When `instanceid` is NOT null (update_group)

1. Go to the Accredible activity edit page.
2. Change a different "Cohort name (from dashboard)" and save.
3. Check if the Accredible activity is linked to the selected _Group_ in Accredible.

#### When `instanceid` is null (create_group)

1. Create a new Accredible activity.
2. Check if the new Accredible activity is successfully created as well as the _Group_ in Accredible.

### CODE REVIEW

**Review by naveenbalachandar — CHANGES_REQUESTED** [review]
Screenshot 2021-09-28 at 7 31 18 PM
@kei178 The code-checker gives out some errors for groups_test.php
Could you please take a look at this as well.

File: classes/local/groups.php [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        return $groups;
+    }
+
+    /**
+     * List all of the issuer's templates
+     *
+     * @return array[stdClass] $templates
+     */
+    public function get_templates() {
+        $response = $this->apiRest->search_groups(10000, 1);
```
[the remark this answers is not in the document]
Comment (kei178) [reply classes/local/groups.php:84]: 'search_groups' has already been covered by existing unit tests.

**Comment by kei178** [comment]
@naveenbalachandar - I added unit tests for the group-related functions in apiRest.php. I also updated the coding style in groups_test.php.

Screen Shot 2021-09-29 at 10 08 44

**Comment by kei178** [comment]
@naveenbalachandar - Fixed the coding style error in apiRest_test.php.

**Review by naveenbalachandar — APPROVED** [review]
LGTM!

**Pull request merged**